fiftyone/core/collections.py

```
@@ -11538,40 +11556,270 @@ async def _async_aggregate(self, aggregations, maxTimeMS=None):
+        }
+
+        view_pipeline = self._pipeline()
+        for stage in view_pipeline:
+            if _PARTITION_UNSAFE_STAGES & stage.keys():
+                return 1
+
+            # $limit is only worth parallelizing when the limit is large
+            # enough that the post-limit work benefits from partitioning;
+            # otherwise every partition redundantly scans to the same limit
+            if "$limit" in stage:
+                if stage["$limit"] < _PARALLEL_SAMPLES_PER_PARTITION:
```

Ruff (0.15.9)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)